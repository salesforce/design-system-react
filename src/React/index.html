<!doctype html public "library">
<meta charset=utf-8/>
<head>
	<title>Selectlist</title>
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
</head>
<body>
<script src="/dist/react/build.js"></script>
</body>
</html>