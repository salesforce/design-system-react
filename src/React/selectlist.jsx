//test

var React = require('react');
import {SelectlistCore} from "../SelectlistCore";
var DropdownButton = require('react-bootstrap/lib/DropdownButton');
var MenuItem = require('react-bootstrap/lib/MenuItem');

var DATA = {
	selected: 'None selected',
	id: 'selectlist1',
	items: [
		{ id: 0, name: 'tacos', type: 'mexican' },
		{ id: 1, name: 'burrito', type: 'mexican' },
		{ id: 2, name: 'tostada', type: 'mexican' },
		{ id: 3, name: 'hush puppies', type: 'southern' }
	]
};

var menuItemSelected = (eventKey, href, target)=> {
	DATA.selected = DATA.items[eventKey].name;
};
//Selectlist.handleMenuItemClicked
var menuItems = DATA.items.map( (menuItem) => {
	return <MenuItem eventKey={menuItem.id}>{menuItem.name}</MenuItem>;
});

var Selectlist = React.createClass({
	getInitialState: function() {
		return {
			selectedKey: undefined
		};
	},

	handleMenuItemClicked: function(eventKey, href, target) {
		this.setState({
			selectedKey: eventKey
		});
	},

	menuItems: function() {
		return DATA.items.map( (menuItem) => {
			return <MenuItem eventKey={menuItem.id} onSelect={this.handleMenuItemClicked}>{menuItem.name}</MenuItem>;
		});
	},

	selected: function() {
		return typeof(this.state.selectedKey) === 'undefined' ? 'None Selected' : DATA.items[this.state.selectedKey].name;
	},

	render () {
		return (
			<div>
				<h1>SelectList</h1>
				<ul>{this.menuItems}</ul>
				<DropdownButton title={this.selected()} key={DATA.id}>{this.menuItems()}</DropdownButton>
			</div>
		);
	}
});

React.render(<Selectlist/>, document.body, () => {});