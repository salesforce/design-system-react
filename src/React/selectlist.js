// SELECTLIST CONTROL

// core
import {Landmark} from '../Landmark';
import {SelectlistCore} from '../SelectlistCore';

// framework specific
import React from 'react';
import DropdownButton from 'react-bootstrap/lib/DropdownButton';
import MenuItem from 'react-bootstrap/lib/MenuItem';

export var Selectlist = React.createClass(Object.assign({}, SelectlistCore, {	
	propTypes: {
		disabled: React.PropTypes.bool,
		selection: React.PropTypes.number,
		collection: React.PropTypes.array
	},
	
	getInitialState () {
		return this.__getInitialState();
	},
		
	menuItems () {
		return this.props.collection.map((menuItem) => {
			return <MenuItem eventKey={menuItem.id} onSelect={this.handleMenuItemClicked}>
					{menuItem.name}
				</MenuItem>;
		});
	},
	
	componentWillReceiveProps(nextProps) {
		this.__initializeOptions(nextProps);
	},
	
	render () {
		var selection = this.getSelection();
		
		return (
			<DropdownButton className={this.cssClasses.CONTROL} disabled={this.state.disabled} title={selection ? selection.name : 'None selected'} key={this.props.id}>{this.menuItems()}</DropdownButton>
		);
	},
	
	componentWillMount () {
		this.elements = {
			wrapper: {
				toggleClass: function (cssClass, state) {
					Landmark.log(cssClass, state);
				}
			}
		}
		
		this.__constructor(this.props);
	},
	
	handleMenuItemClicked (eventKey, href, target) {
		this.setSelectionByKey('id', eventKey);
	}
}));