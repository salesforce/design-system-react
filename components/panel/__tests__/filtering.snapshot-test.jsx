/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ReactDOMServer from 'react-dom/server';
import jsBeautify from 'js-beautify';

import PanelFiltering from '../__examples__/filtering';

test('Panel Filtering Default Snapshot', () => {
	const domTree = toJson(shallow(<PanelFiltering />));
	expect(domTree).toMatchSnapshot();
});

test('Panel Filtering Default HTML Snapshot', () => {
	const domTree = String(
		jsBeautify.html(ReactDOMServer.renderToStaticMarkup(<PanelFiltering />), {
			indent_size: 2,
		}),
		'utf-8',
	);
	expect(domTree).toMatchSnapshot();
});
