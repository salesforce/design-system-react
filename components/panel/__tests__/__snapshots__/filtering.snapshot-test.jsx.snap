// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel Filtering Default Snapshot DOM Snapshot 1`] = `
<div
  className="slds-panel slds-grid slds-grid_vertical slds-nowrap slds-panel_filters"
>
  <div
    className="slds-form_stacked slds-grow slds-scrollable_y slds-grid slds-grid_vertical"
  >
    <div
      className="slds-filters"
    >
      <div
        className="slds-filters__header slds-grid slds-has-divider_bottom-space"
      >
        <h2
          className="slds-align-middle slds-text-heading_small"
        >
          Filter
        </h2>
        <button
          className="slds-button slds-button_icon slds-button_icon-bare slds-button_icon-small slds-col_bump-left"
          disabled={false}
          onClick={[Function]}
          title="Close Panel"
          type="button"
        >
          <svg
            aria-hidden="true"
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#forward"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Close Panel
          </span>
        </button>
      </div>
      <div
        className="slds-filters__body"
      >
        <ol
          className="slds-list_vertical slds-list_vertical-space"
        >
          <li
            className="slds-item slds-hint-parent"
          >
            <div
              className="slds-filters__item slds-grid slds-grid_vertical-align-center"
            >
              <div
                className="slds-grow"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <button
                  className="slds-button_reset slds-grow slds-has-blur-focus"
                  id="sample-panel-filtering-show-me"
                  onClick={[Function]}
                  onFocus={null}
                  onMouseEnter={null}
                  onMouseLeave={null}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="slds-assistive-text"
                  >
                    Edit filter:
                  </span>
                  <p
                    className="slds-text-body_small"
                  >
                    Show Me
                  </p>
                  <p>
                    All Products
                  </p>
                </button>
              </div>
            </div>
          </li>
        </ol>
        <h3
          className="slds-text-body_small slds-m-vertical_x-small"
        >
          Matching all these filters
        </h3>
        <ol
          className="slds-list_vertical slds-list_vertical-space"
        >
          <li
            className="slds-item slds-hint-parent"
          >
            <div
              className="slds-filters__item slds-grid slds-grid_vertical-align-center"
            >
              <div
                className="slds-grow"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <button
                  className="slds-button_reset slds-grow slds-has-blur-focus"
                  id="sample-panel-filtering-created-date"
                  onClick={[Function]}
                  onFocus={null}
                  onMouseEnter={null}
                  onMouseLeave={null}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="slds-assistive-text"
                  >
                    Edit filter:
                  </span>
                  <p
                    className="slds-text-body_small"
                  >
                    Created Date
                  </p>
                  <p>
                    equals THIS WEEK
                  </p>
                </button>
              </div>
              <button
                className="slds-button slds-button_icon slds-button_icon-bare slds-button_icon-small"
                disabled={false}
                onClick={[Function]}
                title="Remove Filter: Created Date equals THIS WEEK"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  className="slds-button__icon slds-button__icon_hint"
                >
                  <use
                    xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
                  />
                </svg>
                <span
                  className="slds-assistive-text"
                >
                  Remove Filter: Created Date equals THIS WEEK
                </span>
              </button>
            </div>
          </li>
          <li
            className="slds-item slds-hint-parent"
          >
            <div
              className="slds-filters__item slds-grid slds-grid_vertical-align-center"
            >
              <div
                className="slds-grow"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <button
                  className="slds-button_reset slds-grow slds-has-blur-focus"
                  id="sample-panel-filtering-list-price"
                  onClick={[Function]}
                  onFocus={null}
                  onMouseEnter={null}
                  onMouseLeave={null}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="slds-assistive-text"
                  >
                    Edit filter:
                  </span>
                  <p
                    className="slds-text-body_small"
                  >
                    List Price
                  </p>
                  <p>
                    greater than "500"
                  </p>
                </button>
              </div>
              <button
                className="slds-button slds-button_icon slds-button_icon-bare slds-button_icon-small"
                disabled={false}
                onClick={[Function]}
                title="Remove Filter: List Price greater than \\"500\\""
                type="button"
              >
                <svg
                  aria-hidden="true"
                  className="slds-button__icon slds-button__icon_hint"
                >
                  <use
                    xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
                  />
                </svg>
                <span
                  className="slds-assistive-text"
                >
                  Remove Filter: List Price greater than "500"
                </span>
              </button>
            </div>
          </li>
        </ol>
      </div>
      <div
        className="slds-filters__footer slds-grid slds-shrink-none"
      >
        <button
          className="slds-button_reset slds-text-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Add Filter
        </button>
        <button
          className="slds-button_reset slds-text-link slds-col_bump-left"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Remove All
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Panel Filtering Default Snapshot HTML Snapshot 1`] = `
"<div class=\\"slds-panel slds-grid slds-grid_vertical slds-nowrap slds-panel_filters\\">
  <div class=\\"slds-form_stacked slds-grow slds-scrollable_y slds-grid slds-grid_vertical\\">
    <div class=\\"slds-filters\\">
      <div class=\\"slds-filters__header slds-grid slds-has-divider_bottom-space\\">
        <h2 class=\\"slds-align-middle slds-text-heading_small\\">Filter</h2><button class=\\"slds-button slds-button_icon slds-button_icon-bare slds-button_icon-small slds-col_bump-left\\" title=\\"Close Panel\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#forward\\"></use></svg><span class=\\"slds-assistive-text\\">Close Panel</span></button></div>
      <div
        class=\\"slds-filters__body\\">
        <ol class=\\"slds-list_vertical slds-list_vertical-space\\">
          <li class=\\"slds-item slds-hint-parent\\">
            <div class=\\"slds-filters__item slds-grid slds-grid_vertical-align-center\\">
              <div class=\\"slds-grow\\" style=\\"display:inline-block\\"><button class=\\"slds-button_reset slds-grow slds-has-blur-focus\\" type=\\"button\\" id=\\"sample-panel-filtering-show-me\\" tabindex=\\"0\\"><span class=\\"slds-assistive-text\\">Edit filter:</span><p class=\\"slds-text-body_small\\">Show Me</p><p>All Products</p></button></div>
            </div>
          </li>
        </ol>
        <h3 class=\\"slds-text-body_small slds-m-vertical_x-small\\">Matching all these filters</h3>
        <ol class=\\"slds-list_vertical slds-list_vertical-space\\">
          <li class=\\"slds-item slds-hint-parent\\">
            <div class=\\"slds-filters__item slds-grid slds-grid_vertical-align-center\\">
              <div class=\\"slds-grow\\" style=\\"display:inline-block\\"><button class=\\"slds-button_reset slds-grow slds-has-blur-focus\\" type=\\"button\\" id=\\"sample-panel-filtering-created-date\\" tabindex=\\"0\\"><span class=\\"slds-assistive-text\\">Edit filter:</span><p class=\\"slds-text-body_small\\">Created Date</p><p>equals THIS WEEK</p></button></div>
              <button
                class=\\"slds-button slds-button_icon slds-button_icon-bare slds-button_icon-small\\" title=\\"Remove Filter: Created Date equals THIS WEEK\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon_hint\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Remove Filter: Created Date equals THIS WEEK</span></button>
            </div>
          </li>
          <li class=\\"slds-item slds-hint-parent\\">
            <div class=\\"slds-filters__item slds-grid slds-grid_vertical-align-center\\">
              <div class=\\"slds-grow\\" style=\\"display:inline-block\\"><button class=\\"slds-button_reset slds-grow slds-has-blur-focus\\" type=\\"button\\" id=\\"sample-panel-filtering-list-price\\" tabindex=\\"0\\"><span class=\\"slds-assistive-text\\">Edit filter:</span><p class=\\"slds-text-body_small\\">List Price</p><p>greater than &quot;500&quot;</p></button></div>
              <button
                class=\\"slds-button slds-button_icon slds-button_icon-bare slds-button_icon-small\\" title=\\"Remove Filter: List Price greater than &quot;500&quot;\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon_hint\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Remove Filter: List Price greater than &quot;500&quot;</span></button>
            </div>
          </li>
        </ol>
    </div>
    <div class=\\"slds-filters__footer slds-grid slds-shrink-none\\"><button class=\\"slds-button_reset slds-text-link\\" type=\\"button\\">Add Filter</button><button class=\\"slds-button_reset slds-text-link slds-col_bump-left\\" type=\\"button\\">Remove All</button></div>
  </div>
</div>
</div>"
`;
